src/server/services/meta_service.cc

```diff
@@ -24,47 +24,27 @@ limitations under the License.
 
 namespace vineyard {
 
-namespace meta_tree {
-void decode_value(const std::string& str, NodeType& type, std::string& value) {
-  if (str[0] == 'v') {
-    type = NodeType::Value;
-    value = str.substr(1);
-  } else if (str[0] == 'l') {
-    type = NodeType::Link;
-    value = str.substr(1);
-  } else {
-    type = NodeType::InvalidType;
-    value.clear();
-  }
-}
-
-void encode_value(NodeType type, const std::string& value, std::string& str) {
-  str.clear();
-  if (type == NodeType::Value) {
-    str.resize(value.size() + 1);
-    str[0] = 'v';
-    memcpy(&str[1], value.c_str(), value.size());
-  } else if (type == NodeType::Link) {
-    str.resize(value.size() + 1);
-    str[0] = 'l';
-    memcpy(&str[1], value.c_str(), value.size());
-  }
-}
-}  // namespace meta_tree
-
 std::shared_ptr<IMetaService> IMetaService::Get(vs_ptr_t ptr) {
   return std::shared_ptr<IMetaService>(new EtcdMetaService(ptr));
 }
 
-void IMetaService::incRef(std::string const& key, std::string const& value) {
+void IMetaService::incRef(std::string const& key, json const& value) {
+  if (!value.is_string()) {
+    return;
+  }
   std::vector<std::string> vs;
-  boost::algorithm::split(vs, key, [](const char c) { return c == '.'; });
+  boost::algorithm::split(vs, key, [](const char c) { return c == '/'; });
+  if (vs[0].empty()) {
```

Comment:
Is it possible that vs.size() == 0?

Reply:
The vector cannot be empty, since `boost::algorithm::split` on empty string yields `{""}`.